controller/volume_controller.go

```diff
@@ -2999,6 +3000,112 @@ func (vc *VolumeController) ResolveRefAndEnqueue(namespace string, ref *metav1.O
 	vc.enqueueVolume(volume)
 }
 
+// createRecurringJobNotExistAndLabel add recurring jobs and group back to a restoring volume and create recurring jobs if not exist.
+func createRecurringJobNotExistAndLabel(log *logrus.Entry, ds *datastore.DataStore, v *longhorn.Volume, jobName string, job *longhorn.RecordVolumeRecurringJob) error {
+	if existJob, err := ds.GetRecurringJob(jobName); err != nil {
+		if !apierrors.IsNotFound(err) {
+			log.WithError(err).Warnf("Failed to get the recurring job %v", jobName)
+			return err
+		}
+		// create the recurring job if not exist
+		log.Debugf("Create the recurring job %v when restoring recurring jobs", jobName)
+		if _, err := ds.CreateRecurringJob(&longhorn.RecurringJob{ObjectMeta: metav1.ObjectMeta{Name: jobName}, Spec: job.JobSpec}); err != nil {
+			log.WithError(err).Warnf("Failed to create a recurring job %v", jobName)
+			return err
+		}
+	} else {
+		// create the recurring job if job name is used and configuration is different.
+		if !reflect.DeepEqual(existJob.Spec, job.JobSpec) {
```

Comment:
nit: 5 layers of indentation. Suggest refactoring and reducing to 3 layers and make it easier to read.

Reply:
Modified. Separated into a function.